WordPress/Classes/ViewRelated/Post/PublishSettingsViewController.swift

```diff
@@ -0,0 +1,179 @@
+
+    var date: Date? {
+        switch state {
+        case .scheduled(let date), .published(let date):
+            return date
+        case .immediately:
+            return nil
+        }
+    }
+
+    mutating func setDate(_ date: Date?) {
+        if let date = date {
+            state = .scheduled(date)
+        } else {
+            state = .immediately
+        }
+
+        post.dateCreated = date
+    }
+}
+
+@objc class PublishSettingsController: NSObject, SettingsController {
+
+    @objc class func viewController(post: AbstractPost) -> ImmuTableViewController {
+        let controller = PublishSettingsController(post: post)
+        let viewController = ImmuTableViewController(controller: controller)
+        return viewController
+    }
+
+    var noticeMessage: String?
+
+    let title = NSLocalizedString("Publish", comment: "Title for the publish settings view")
+
+    var immuTableRows: [ImmuTableRow.Type] {
+        return [
+            EditableTextRow.self
+        ]
+    }
+
+    static var dateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .long
+        dateFormatter.timeStyle = .short
+        return dateFormatter
+    }()
+
+    private var viewModel: PublishSettingsViewModel
+
+    init(post: AbstractPost) {
+        viewModel = PublishSettingsViewModel(post: post)
+
+        let siteTimezoneOffset = viewModel.timeZone?.gmtOffset ?? 0
+        let deviceTimezoneOffset = TimeZone.current.secondsFromGMT()/60/60
+        if siteTimezoneOffset != Float(deviceTimezoneOffset) {
+            noticeMessage = NSLocalizedString("Site time zone differs from device time zone", comment: "Notice that time zones are different when scheduling")
+        }
+    }
+
+    func tableViewModelWithPresenter(_ presenter: ImmuTablePresenter) -> ImmuTable {
+        return mapViewModel(viewModel, presenter: presenter)
+    }
+
+    func refreshModel() {
+        // Don't need to refresh the model
+    }
+
+    func mapViewModel(_ viewModel: PublishSettingsViewModel, presenter: ImmuTablePresenter) -> ImmuTable {
+
+        let rows: [ImmuTableRow] = viewModel.cells.map { cell in
+            switch cell {
+            case .dateTime:
+                return EditableTextRow(
+                    title: NSLocalizedString("Date and Time", comment: "Date and Time"),
+                    value: viewModel.date?.shortStringWithTime() ?? NSLocalizedString("Immediately", comment: "Undated post time label"),
+                    action: presenter.present(dateTimeCalendar(model: viewModel))
+                )
+            }
+        }
+
+        let footerText: String?
+
+        if let date = viewModel.date {
+            let publishedOnString = PublishSettingsController.dateFormatter.string(from: date)
+            let offsetLabel = viewModel.timeZone?.label ?? "Unknown Offset"
+            footerText = "Post will be published on \(publishedOnString) in your site time zone (\(offsetLabel))"
```

Comment:
The `footerText` should be localized (including the optional "Unknown Offset"). 

Also worth noting, we don't use string interpolation with localized text since it causes issues for the translators. We typically use `String.localizedStringWithFormat` to accomplish both.